src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Kepler/BrouwerLyddaneMean.cpp

```diff
@@ -0,0 +1,291 @@
+{
+    const COE coe = this->toCOE();
+    return coe.getCartesianState(aGravitationalParameter, aFrameSPtr);
+}
+
+BrouwerLyddaneMean::BrouwerLyddaneMean(
+    const Length &aSemiMajorAxis,
+    const Real &anEccentricity,
+    const Angle &anInclination,
+    const Angle &aRaan,
+    const Angle &anAop,
+    const Angle &aMeanAnomaly
+)
+    : COE(aSemiMajorAxis, anEccentricity, anInclination, aRaan, anAop, aMeanAnomaly, COE::AnomalyType::Mean)
+{
+}
+
+Vector6d BrouwerLyddaneMean::Cartesian(
+    const COE::CartesianState &aCartesianState,
+    const Derived &aGravitationalParameter,
+    std::function<Vector6d(const Vector6d &)> toCOEVector
+)
+{
+    const COE coe = COE::Cartesian(aCartesianState, aGravitationalParameter);
+
+    bool possibleInaccuracyLongWritten = false;
+    bool inaccuracyCriticalAngleWritten = false;
+
+    if ((coe.getEccentricity() > 0.99) || (coe.getEccentricity() < 0.0))
+    {
+        throw ostk::core::error::RuntimeError(
+            String::Format("BrouwerLyddaneMean is applicable only if 0.0 < Ecc: [{}] < 0.99.", coe.getEccentricity())
+        );
+    }
+
+    const Real perigee = coe.getPeriapsisRadius().inMeters();
+    if (perigee < 3000000.0)
+    {
+        throw ostk::core::error::RuntimeError(
+            String::Format("BrouwerLyddaneMean is applicable only if Perigee radius: [{}] > 3000000.0 m.", perigee)
+        );
+    }
+    if (perigee < 6378000.0)
+    {
+        if (!possibleInaccuracyLongWritten)
+        {
+            std::cout << "Warning: For BrouwerLyddaneMean, when RadPer < 6378km, there is a "
+                         "possible inaccuracy due to singularity "
+                         "related with inside-of-earth orbit.\n";
+            possibleInaccuracyLongWritten = true;
+        }
+    }
+
+    const Real inclination_degrees = coe.getInclination().inDegrees();
+    if (inclination_degrees > 180.0)
+    {
+        throw ostk::core::error::RuntimeError(String::Format(
+            "BrouwerLyddaneMean is applicable only if inclination: [{}] < 180.0 degrees.", inclination_degrees
+        ));
+    }
+
+    if ((58.80 < inclination_degrees && inclination_degrees < 65.78) ||
+        (114.22 < inclination_degrees && inclination_degrees < 121.2))
+    {
+        if (!inaccuracyCriticalAngleWritten)
+        {
+            std::cout << "Warning: For BrouwserMeanLong, when 58.80 DEG < "
+                         "INC < 65.78 DEG, or 114.22 DEG "
+                         "< INC < 121.2DEG, there is a possible inaccuracy due to "
+                         "singularity related with critical angle.\n";
+            inaccuracyCriticalAngleWritten = true;
+        }
+    }
+
+    Vector6d cartesian;
+    cartesian.segment(0, 3) = aCartesianState.first.accessCoordinates();
+    cartesian.segment(3, 3) = aCartesianState.second.accessCoordinates();
+
+    Vector6d coeVector = coe.getVector(COE::AnomalyType::Mean);
+
+    Integer pseudostate = 0;
+    if (coeVector[2] > 3.0543261909900763)
```

Comment:
Is the logic/numerical values taken from GMAT here? Mabe let's add a ref link?

Reply:
Yeah, I've added a reference at top, will put a comment on the degrees number.

Reply:
Sweet!